packages/fuels-code-gen/src/program_bindings/utils.rs

```diff
@@ -19,6 +19,7 @@ use crate::{
 pub(crate) struct Component {
     pub field_name: Ident,
     pub field_type: ResolvedType,
+    pub wrap_into: bool,
```

Comment:
I think this field should have some documentation as to why it's there and what purposes it serves. Also I am not sure I understand, if this is only for `ContractId` and `Address` types, why have this field at all and not just use a boolean `Component::wraps_into()` method? And you would have an enum that hold all the strings that need to be wrapped into (for now, just `Address` and `ContractId`).

Reply:
resolving as the code changed. See: https://github.com/FuelLabs/fuels-rs/pull/967#issuecomment-1556131371